app/models/environment.rb

```diff
@@ -96,6 +128,23 @@ def installation
     repository.installation
   end
 
+  def check_config!
+    fail("This repository is configured from .slashdeploy.yaml") if repository && repository.config?
+  end
+
+  def config?
+    config.present?
+  end
+
+  def continuous_delivery_config?
+    config? && config.continuous_delivery.present?
+  end
+
+  def config
+    return unless repository && repository.config?
+    repository.config.environments[name]
```

Comment:
I guess this is a style thing. There's two ways people generally code conditions:

**if/else**

```ruby
if thing
  this
else
  that
end
```

**early return**

```ruby
return this if thing
that
```

In general, I prefer that latter (early return) just because it's less cognitive load if a method is large (if it returns, you can just stop reading). In this case, the early return is really a "guard".